dags/landsat_scenes_sync/landsat_scenes_gap_report.py

```diff
@@ -196,6 +205,59 @@ def request_usgs_api(url: str):
                 yield future.result()
 
 
+def check_for_stac_files(dest_paths):
+    """
+    If there is a missing stac, either ST or SR, and the stac is present in USGS, consider a not transfer scene
+    """
+    def find_stac_files(prefix):
+
+        s3 = S3(conn_id=CONN_LANDSAT_SYNC)
+
+        scene = prefix.split('/')[-2]
+
+        # Checking SR File in Africa
+        sr_file = s3.list_objects(
+            bucket_name=LANDSAT_SYNC_BUCKET_NAME,
+            region=REGION,
+            prefix=f"{prefix}{scene}_SR_stac.json"
+        )
+
+        if not sr_file.get("Contents"):
+            # Checking SR file in USGS
+            sr_usgs_file = s3.list_objects(
+                bucket_name=USGS_S3_BUCKET_NAME,
+                region=USGS_AWS_REGION,
+                prefix=f"{prefix}{scene}_SR_stac.json"
+            )
+            if sr_usgs_file.get("Contents"):
+                return
```

Comment:
Why check USGS?

Reply:
Because sometimes USGS does not have the key and in this case, we are ok with not have that either.
If I don`t check and they don`t have that, we will end up in a deadlock, which we'll always have this key as missing and will try to transfer.